components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -140,10 +143,26 @@ public IdentityProvider getResidentIdP(String tenantDomain)
         String samlAuthnRequestsSigningEnabled;
 
         openIdUrl = IdentityUtil.getProperty(IdentityConstants.ServerConfig.OPENID_SERVER_URL);
-        samlECPUrl = IdentityUtil.resolveURL(IdentityUtil.getProperty(IdentityConstants.ServerConfig.SAML_ECP_URL),
-                true, true);
-        samlArtifactUrl = IdentityUtil.resolveURL(IdentityUtil.getProperty(IdentityConstants.ServerConfig
-                .SSO_ARTIFACT_URL), true, true);
+
+        if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+            try {
+                samlECPUrl = ServiceURLBuilder.create().addPath(IdPManagementConstants.SAML_ECP_URL).build()
+                        .getAbsoluteURL();
+            } catch (URLBuilderException e) {
+                throw IdentityRuntimeException
+                        .error("Error while building URL: " + IdPManagementConstants.SAML_ECP_URL, e);
```

Comment:
Should this be throwing a runtime exception ? Can use IdentityProviderManagementException right